### PR TITLE
[chore][jmxreceiver] Update flaky JMX receiver integration test

### PR DESCRIPTION
**Pull request by mrsillydog**
**Description:** Update the JMX receiver integration test to consistently pass on jar versions that are currently supported.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5874

**Testing:** Added testing for the opentelemetry-jmx-metrics-1.26.0-alpha.jar and opentelemetry-jmx-metrics-1.10.0-alpha.jar. Removed testing for the now-unsupported opentelemetry-java-contrib-jmx-metrics-1.0.0-alpha.jar and opentelemetry-jmx-metrics/1.4.0-alpha/opentelemetry-jmx-metrics-1.4.0-alpha.jar.

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: receiver/jmxreceiver/internal/subprocess/subprocess.go [hunk 1]
```diff
@@ -131,6 +131,8 @@ func (subprocess *Subprocess) Shutdown(ctx context.Context) error {
 	}
 	t := time.NewTimer(timeout)
 
+	subprocess.cancel()
```
Comment (djaglowski) [inline receiver/jmxreceiver/internal/subprocess/subprocess.go:134]: Is this a bug fix?

Comment (mrsillydog) [reply receiver/jmxreceiver/internal/subprocess/subprocess.go:134]: Yes - without it, this test generates a new rogue java process each time that it's run, without cleaning up after itself. I assume the same is true of the receiver itself.

File: receiver/jmxreceiver/testdata/integration/expected.1.10.0.yaml [hunk 2]
```diff
@@ -0,0 +1,270 @@
+resourceMetrics:
```
[the remark this answers is not in the document]
Comment (mrsillydog) [reply receiver/jmxreceiver/testdata/integration/expected.1.10.0.yaml:1]: The version naming of the jars is identically inconsistent, unfortunately. See receiver/jmxreceiver/supporter_jars. Worth changing in both places? Or is there a reason why -alpha is left off of the early jar versions?

File: receiver/jmxreceiver/internal/subprocess/subprocess.go (hunk 1 above)
Comment (djaglowski) [reply receiver/jmxreceiver/internal/subprocess/subprocess.go:134]: This is probably worth submitting as a separate PR, with a changelog entry.

Comment (mrsillydog) [reply receiver/jmxreceiver/internal/subprocess/subprocess.go]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23051

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**